go/arrow/memory/buffer.go

```diff
@@ -33,6 +33,10 @@ type Buffer struct {
 	parent *Buffer
 }
 
+func NewBufferWithAllocator(data []byte, mem Allocator) *Buffer {
```

Comment:
There isn't really a way to expose it *only* in `importAllocator.go` because we don't expose any ability to set/change the allocator of a buffer (nor do we want to) essentially the difference between *this* method and `NewResizableBuffer` is that `mutable` will be marked as false.

Reply:
Ah, ok. Can you add a comment or docstring explaining the semantics?

Reply:
added